### PR TITLE
v31tov24: Fix translation without a proxy

### PR DESCRIPTION
**Pull request by cgwalters**
I hit this in cosa trying to downtranslate to v2, we crash
without a proxy set.

### CODE REVIEW

File: translate/v31tov24/v31tov24.go [hunk 1]
```diff
@@ -57,6 +57,15 @@ func Translate(cfg types.Config) (old.Config, error) {
 	// named by the path
 	fsList := generateFsList(cfg.Storage.Filesystems)
 
+	var httpProxy string
+	if cfg.Ignition.Proxy.HTTPProxy != nil {
+		httpProxy = *cfg.Ignition.Proxy.HTTPProxy
```
Comment (cgwalters) [inline translate/v31tov24/v31tov24.go:62]: I so miss the Rust `Option` + `unwrap_or_else()`...

File: go.sum [hunk 2]
```diff
@@ -13,6 +13,7 @@ github.com/coreos/ignition v0.35.0 h1:UFodoYq1mOPrbEjtxIsZbThcDyQwAI1owczRDqWmKk
 github.com/coreos/ignition v0.35.0/go.mod h1:WJQapxzEn9DE0ryxsGvm8QnBajm/XsS/PkrDqSpz+bA=
 github.com/coreos/ignition/v2 v2.3.0 h1:TK+STbzVe6KZp4tQ2IaNSRMiWX4/diNngep1F7tP7Zk=
 github.com/coreos/ignition/v2 v2.3.0/go.mod h1:85dmM/CERMZXNrJsXqtNLIxR/dn8G9qlL1CmEjCugp0=
+github.com/coreos/ignition/v2 v2.7.0 h1:JCKxJllVtnk1lQY1uisxrtFSHG5L2NI1LRzc8wBEk84=
```
Comment (bgilbert) [inline go.sum:16]: Stray line?

File: translate_test.go [hunk 3]
```diff
@@ -1274,7 +1274,18 @@ func TestTranslate3_1to2_2(t *testing.T) {
...
 
 func TestTranslate3_1to2_4(t *testing.T) {
-	res, err := v31tov24.Translate(nonexhaustiveConfig3_1)
+	emptyConfig := types3_1.Config{
+		Ignition: types3_1.Ignition{
+			Version: "3.1.0",
+		},
+	}
+
+	res, err := v31tov24.Translate(emptyConfig)
```
Comment (bgilbert) [inline translate_test.go:1283]: Let's add this to the other translator tests as well.

File: translate/v31tov24/v31tov24.go (hunk 1 above)
Comment (bgilbert) [reply translate/v31tov24/v31tov24.go:62]: We have `util.StrV()`.

File: go.sum (hunk 2 above)
Comment (cgwalters) [reply go.sum]: https://github.com/microsoft/vscode-go/issues/2730
Don't other people hit this?  Not sure it's worth a SOP to explicitly split these out.  But anyways, reverted for now.

File: translate/v31tov24/v31tov24.go (hunk 1 above)
Comment (cgwalters) [reply translate/v31tov24/v31tov24.go]: OK thanks, definitely better.

**Pull request merged**